src/store/actions.ts

```diff
@@ -489,14 +489,18 @@ export const resubmit = async ({ commit, getters }): Promise<boolean> => {
 
   // add resubmit NR number (for internal use only - API ignores it)
   nrTypeData['resubmitNrNum'] = nrData['nrNum']
-
+  // check against entity types if it should have designation or not : if length returns to be 0 then NR should not have any designation otherwise work as getNameDesignation
+  let flag = false
+  if (Designations[nrData['entity_type_cd']].words.length === 0) {
+    flag = true
+  }
```

Comment:
I'd name the flag something useful.

Also, you can simplify this...
```
const flag = (your boolean statement)
```

Reply:
Thanks for the very useful comment!

Reply:
Thanks for the update. Sorry for being so picky. Your code looks good and I've approved this PR but I think you have to wait a couple of days before merging it (check with Omid).